metrics.go

```diff
@@ -15,118 +15,29 @@
 package metrics
 
 import (
-	"math"
 	"net/http"
 
-	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-const (
-	// MetricsEndpoint is the port to export metrics on
-	MetricsEndpoint = ":8080"
-)
-
-var (
-	metricTotalAWSAccounts = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_aws_accounts",
-		Help: "Report how many accounts have been created in AWS org",
-	}, []string{"name"})
-	metricTotalAccountCRs = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_account_crs",
-		Help: "Report how many account CRs have been created",
-	}, []string{"name"})
-	metricTotalAccountCRsUnclaimed = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_accounts_crs_unclaimed",
-		Help: "Report how many account CRs are unclaimed",
-	}, []string{"name"})
-	metricTotalAccountCRsClaimed = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_accounts_crs_claimed",
-		Help: "Report how many account CRs are claimed",
-	}, []string{"name"})
-	metricTotalAccountCRsFailed = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_accounts_crs_failed",
-		Help: "Report how many account  CRs are failed",
-	}, []string{"name"})
-	metricTotalAccountClaimCRs = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_total_aws_account_claim_crs",
-		Help: "Report how many account claim CRs have been created",
-	}, []string{"name"})
-	metricPoolSizeVsUnclaimed = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "aws_account_operator_pool_size_vs_unclaimed",
-		Help: "Report the difference between the pool size and the number of unclaimed account CRs",
-	}, []string{"name"})
-
-	metricsList = []prometheus.Collector{
-		metricTotalAWSAccounts,
-		metricTotalAccountCRs,
-		metricTotalAccountCRsUnclaimed,
-		metricTotalAccountCRsClaimed,
-		metricTotalAccountCRsFailed,
-		metricTotalAccountClaimCRs,
-		metricPoolSizeVsUnclaimed,
-	}
-)
-
 // StartMetrics register metrics and exposes them
-func StartMetrics() {
+func StartMetrics(list []prometheus.Collector, fn recordmetrics, path string, port string) {
```

Comment:
We should put this in a struct that has a list of prometheus metrics defined in it. 

I think I'm understanding what you're trying to achieve with the register function as a parameter. However, I think that just calling that directly is ok, as the logic of registering with a prometheus server will be more or less the same everywhere. 

If we make a struct to hold all the data, the metrics path and port could go in there too.

Reply:
The inputs which we would be getting from the user are:
list of metrics to be registered, port Name, Path, and a user-defined function on how the metrics are to be collected. So, shall I include the first three functions in the struct?